src/entries/popup/pages/home/NFTs/NFTCollectionSection.tsx

```diff
@@ -49,16 +54,57 @@ export function NFTCollectionSection({
   const { isWatchingWallet } = useWallets();
   const sections = useNftsStore.use.sections();
   const toggleGallerySectionOpen = useNftsStore.use.toggleGallerySectionOpen();
+  const hidden = useNftsStore.use.hidden();
+  const hiddenNftsForAddress = hidden[address] || {};
   const sectionsForAddress = sections[address] || {};
-  const collectionId = section?.collection?.collection_id;
-  const collectionVisible = collectionId && sectionsForAddress[collectionId];
+  const collectionId = collection?.collection_id;
+  const totalCopiesOwned = collection?.distinct_nfts_owned;
+  const isHiddenSection =
+    collection.collection_details.description === '_hidden';
+  const shouldHideHiddenSection = !Object.values(
+    hiddenNftsForAddress || {},
+  ).filter(Boolean).length;
+  const collectionVisible = !!(
+    collectionId &&
+    sectionsForAddress[isHiddenSection ? '_hidden' : collectionId]
+  );
+  const {
+    data,
+    hasNextPage,
+    isFetching,
+    isFetchingNextPage,
+    fetchNextPage,
+    isLoading,
+  } = useNftsForCollection(
+    {
+      address,
+      collectionId,
+      collectionChains: (isHiddenSection
+        ? getSimpleHashSupportedChainNames()
+        : collection?.collection_details?.chains) as ChainName[],
+    },
+    {
+      enabled: collectionVisible,
+    },
+  );
+  const nfts = selectNfts(data)?.filter((n) => {
+    if (isHiddenSection) {
+      return hiddenNftsForAddress[n.uniqueId];
+    }
+    return !hiddenNftsForAddress[n.uniqueId];
+  });
   const setCollectionVisible = useCallback(() => {
     toggleGallerySectionOpen({
       address,
-      collectionId: collectionId || '',
+      collectionId: isHiddenSection ? '_hidden' : collectionId || '',
```

Comment:
how do you hide? i tried to test this but i don't see any hide button

Reply:
You may have been looking at a wallet you were watching.